Proper Fix for 'Package cl is deprecated' warning

ifiinlist opened the pull request:
The package `cl` is deprecated. Emacs now has a built in equivalent `cl-lib` since Emacs v24.3. This commit properly implements those cl-lib equivalent functions. Also I changed the minimum Emacs version in the README to 24.3 . This PR also closes #125 and #127

ifiinlist committed:
replacing cl functions with cl-lib equivalents

The package 'cl' is deprecated. Emacs now has a built in
equivalent 'cl-lib'. This commit properly implements those
cl-lib equivalent functions.

ifiinlist committed:
Updating the supported Emacs version to 24.3

The package 'cl-lib' is not installed by default on
Emacs < v24.3 . That's why I'm increasing the Emacs
version.

Zerogaku commented:
@ifiinlist Hey, I recently applied this pr to my fork, but it seems to not have fixed the issue, the warning will still show.

ifiinlist commented:
@Zerogaku It's working fine on my end. I also tried out your fork. It didn't show the warning.
Have you set up your `load-path` correctly?

Zerogaku commented:
Whoops my bad, yeah doom emacs wasn't loading the forked github repo correctly and was still using the default emmet-mode, after finding the solution in this issue https://github.com/doomemacs/doomemacs/issues/7235#issuecomment-1679479228 the deprecated warning no longer appeared, thanks for the help!

ifiinlist committed:
Fix: expressions not being expanded

This commit uses lexical-binding to get rid of lexical-let.
While this does fix the issue, it generates a lot of byte-compile
warnings. I'll work on the warnings.

smihica commented:
thank you
